frontend/revive-gallery-app/src/components/products/ProductsList.js

```diff
@@ -1,54 +1,57 @@
 import React, { useEffect, useState } from "react";
 import { Link } from "react-router-dom";
+import '../../styles/products/ProductsList.css';
 
 const ProductsList = () => {
+
   const [products, setProducts] = useState([]);
+  const [error, setError] = useState(null);
 
   useEffect(() => {
     // Fetch all the products from the backend API
     fetch("http://localhost:8080/api/product", {
-        method: "GET",
-        headers: {
-            "Content-Type": "application/json",
-        },
+      method: "GET",
+      headers: {
+        "Content-Type": "application/json",
+      },
     })
-    .then((response) => {
-        if (response.ok) {
-            return response.json();
-        } else {
-            console.log("Request failed. Status: " + response.status);
-            throw new Error("Request failed");
+      .then((response) => {
+        if (!response.ok) {
+          throw new Error("Request failed with status: " + response.status);
         }
-    })
-    .then((data) => {
+        return response.json();
+      })
+      .then(async (data) => {
         console.log('Received data:', data);
         setProducts(data.products);
-    })
-    .catch((error) => console.error("Error fetching products:", error));
+      })
+      .catch((error) => {
+        console.error("Error fetching products:", error);
+        setError(error.message);
+      });
   }, []);
 
   return (
     <div>
-      <h1>Marketplace</h1>
-      <table className="product-table">
-        <thead>
-          <tr>
-            <th>Product Name</th>
-            <th>Description</th>
-            <th>Price</th>
-          </tr>
-        </thead>
-        <tbody>
-          {products.map((product) => (
-            <tr key={product._id}>
-              <td>{product.title}</td>
-              <td>{product.description}</td>
-              <td>${product.price}</td>
-              <td><Link to={`/products/details?id=${product._id}`}>{product._id}</Link></td>
-            </tr>
-          ))}
-        </tbody>
-      </table>
+      {error && <p>Error: {error}</p>}
+      <div className="product-container">
+        {products.map((product) => (
+          <div key={product._id} className="product-item">
+            <Link to={`/products/details?id=${product._id}`}>
+              <div className="product-image">
+                <img
+                  src={product.images[0]} 
```

Comment:
@SaiKiran0407 You are directly placing the url in the img src. I am not sure whether it displays the image properly or not? I think we need to get the download urls first. Did it work for you as expected?

Reply:
@anjaliputta The URLs that are stored in the database have already been downloaded. They can be directly placed in img tags. Yes, the images are properly displayed, and it's working as expected.

Reply:
@SaiKiran0407 I got it now. Thank you.